Documentation/MyFirstContribution

```diff
@@ -0,0 +1,584 @@
+to BUILTIN_OBJS, and add `builtin/psuh.o` in the same way. Once you've done so,
+move to the root directory and build simply with `make -j16`. Optionally, add
+the DEVELOPER=1 variable to turn on some additional warnings:
+
+----
+echo DEVELOPER=1 > config.mak
+make -j16
+----
+
+Great, now your new command builds happily on its own. But nobody invokes it.
+Let's change that.
+
+The list of commands lives in `git.c`. We can register a new command by adding
+a cmd_struct to the commands[] array. struct cmd_struct takes a string with the
+command name, a function pointer to the command implementation, and a setup
+option flag. For now, let's keep cheating off of push. Find the line where
+cmd_push is registered, copy it, and modify it for cmd_psuh. Leave the options
+the same as for push.
+
+TODO: what do the options do? How to fix them?
+
+Go ahead and build again. You should see a clean build, so let's kick the tires
+and see if it works. There's a binary you can use to test with in
+`./bin-wrappers`.
+
+----
+./bin-wrappers/git psuh
+----
+
+Check it out! You've got a command! Nice work!  Let's commit this.
+
+----
+git add Makefile builtin.h builtin/psuh.c git.c
+git commit -sm "builtin: add hooks for git-psuh"
+----
+
+Go ahead and inspect your new commit with `git show`. "builtin:" indicates you
+have modified mainly the builtin/ area. The subject line gives readers an idea
+of what you've changed. The signed-off line (-s) indicates that you agree to
+the Developer's Certificate of Origin 1.1 (see the SubmittingPatches [[dco]]
+header). If you wish to add some context to your change, go ahead with
+`git commit --amend`.
+
+=== Implementation
+
+It's probably useful to do at least something besides print out a string. Let's
+start by having a look at everything we get.
+
+Modify your `cmd_psuh` implementation to dump the args you're passed:
+
+----
+	printf("Your args (there are %i):\n", argc);
+	for (int i = 0; i < argc; i++) {
+		printf("%s\n", argv[i]);
+	}
+	printf("Your prefix:\n%s\n", prefix);
+----
+
+As you may expect, there's pretty much just whatever we give on the command
+line, including the name of our command. That's not so helpful. So what other
+context can we get?
+
+Add a line to `#include "config.h"`. Then, add the following bits:
+
+----
+const char *cfg_name;
+
+...
+
+git_config(git_default_config, NULL)
+if (git_config_get_string_const("user.name", &cfg_name) > 0)
+{
+	printf("No name is found in config\n");
+}
+else
+{
+	printf("Your name: %s\n", cfg_name);
+}
+----
+
+git_config(...) will grab the configuration from config files known to Git and
+apply standard precedence rules. git_config_get_string_const(...) will look up
+a specific key ("user.name") and give you the value. There are a number of
+single-key lookup methods like this one; you can see them all (and more info
+about how to use git_config()) in `Documentation/technical/api-config.txt`.
```

Comment:
Maybe this reference to `api-config.tx` could be made a link?

Reply:
Do you think it's better to go through and turn all of the references to other docs into links?